bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/common/WeatherConfig.java

```diff
@@ -61,62 +61,77 @@
     private Map<ProviderName, ProviderConfig> providerConfigs = new HashMap<ProviderName, ProviderConfig>();
     private Map<String, LocationConfig> locationConfigs = new HashMap<String, LocationConfig>();
 
-    private boolean valid = false;
+    private boolean valid;
+    private boolean parseCompleted;
 
     /**
      * Parses and validates the properties in openhab.cfg.
      */
     public void parse(Dictionary<String, ?> properties) throws ConfigurationException {
+        parseCompleted = false;
         valid = false;
-        if (properties != null) {
-            Enumeration<String> keys = properties.keys();
-            while (keys.hasMoreElements()) {
-                String key = keys.nextElement();
-
-                String value = StringUtils.trimToNull((String) properties.get(key));
-                if (StringUtils.startsWithIgnoreCase(key, "apikey")) {
-                    parseApiKey(key, value);
-                } else if (StringUtils.startsWithIgnoreCase(key, "location")) {
-                    parseLocation(key, value);
-                }
+        if (properties == null ) {
```

Comment:
Hi Chris, please format your changes with the Eclipse IDE formatter, so they won't be reformatted on the next commit.